pyms/flask/services/service_discovery.py

```diff
@@ -0,0 +1,82 @@
+import json
+import logging
+import uuid
+
+import requests
+from requests.exceptions import RequestException
+from pyms.constants import LOGGER_NAME
+from pyms.exceptions import ServiceDiscoveryConnectionException
+from pyms.flask.services.driver import DriverService
+
+logger = logging.getLogger(LOGGER_NAME)
+
+CONSUL_SERVICE_DISCOVERY = "consul"
+
+DEFAULT_SERVICE_DISCOVERY = CONSUL_SERVICE_DISCOVERY
+
+
+class ServiceDiscoveryBase:
+    def register_service(self, id_app, host, port, healtcheck_url, app_name):
+        pass
+
+
+class ServiceDiscoveryConsul(ServiceDiscoveryBase):
```

Comment:
Do you plan to implement all the stages of the service discovery? It seems to be some work done, in case you don't want to implement it all:
- https://github.com/bigdatacesga/service-discovery 
- https://github.com/cablehead/python-consul

Reply:
Reimplementing and maintaining the whole Consul API would be A LOT of work. While there is no official one it seems that `consulate` and `python-consul2` (fork of mentioned `python-consul` are the most up to date (consulate doesn't have any release for some time but most functions seem to work as expected).

Reply:
`consulate` is very very good implementation with a intuitive implementation, but as you said @Agalin this package hasn't a up-to-date maintenance. I need to figure out `python-consul2` :+1: 

I reimplementing because i was thinking that only need a register and deregister endpoint in our use case and other options are "over-engineering"

Reply:
Personally I really like consulate. Hoping for a fork or something. 
More important part of the service discovery is accessing the registry - that can be done by consul’s dns service without a dedicated api usage. But maintaining acl tokens api as proposed - or advanced queries for consul service registry - may be a bit too complex to maintain as part of this project. So question is how advanced this service is supposed to be?

Reply:
With your approach, Could be better get access to `consulate` as a client as MVP and if we have an issue create a fork in PyMs of `consulate`, are you agree?

when i mean "client", i mean:

```
    service_discovery:
      client: "consul"
      host: "http://localhost:8500"
      autoregister: true
```

And use this service in all application with `current_app.ms.service_discovery.[all-consulate-methods]`

Reply:
I think it should be under `client`.
```python
current_app.ms.service_discovery.client.[all-consulate-methods]
# generic service retrieval call provided by all implementations, would retrieve an url to access the service - to be used either by user or in other services
current_app.ms.service_discovery.find_service(name, **kwargs)
```